src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -101,15 +102,17 @@ internal sealed class HttpConnectionPool : IDisposable
         /// <summary>Initializes the pool.</summary>
         /// <param name="poolManager">The manager associated with this pool.</param>
         /// <param name="kind">The kind of HTTP connections stored in this pool.</param>
+        /// <param name="socksKind">The kind of SOCKS connection to use in this pool.</param>
         /// <param name="host">The host with which this pool is associated.</param>
         /// <param name="port">The port with which this pool is associated.</param>
         /// <param name="sslHostName">The SSL host with which this pool is associated.</param>
         /// <param name="proxyUri">The proxy this pool targets (optional).</param>
         /// <param name="maxConnections">The maximum number of connections allowed to be associated with the pool at any given time.</param>
-        public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionKind kind, string? host, int port, string? sslHostName, Uri? proxyUri, int maxConnections)
+        public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionKind kind, SocksConnectionKind socksKind, string? host, int port, string? sslHostName, Uri? proxyUri, int maxConnections)
```

Comment:
Maybe a new enum field in HttpConnectionKind is better?

Reply:
I think 2 new values in `HttpConnectionKind` would be better. It would mean we don't pollute the connection pool implementation with socks support in too many places. Also in  https://github.com/dotnet/runtime/blob/56b2fa93da517e3812456d8909454020773a242a/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L1216
we wouldn't have to duplicate across the `Http` and `Https` kinds (looks like the proxy is ignored for http right now?).

We wouldn't want to increase the size of the `HttpConnectionKey` just because of this.